blueocean-dashboard/src/main/js/api/Branch.js

```diff
@@ -29,4 +29,18 @@ export default class Branch extends Pipeline {
         }
         return false;
     }
+
+    /**
+     * Check if two branch names are the same.
+     * Works around encoding mismatch bugs by encoding each branch name and comparing to the other.
+     * @param branchName1
+     * @param branchName2
+     * @returns {boolean}
+     * @private
+     */
+    _compareBranchNames(branchName1, branchName2) {
+        return branchName1 === branchName2 ||
+            encodeURIComponent(branchName1) === branchName2 ||
+            branchName1 === encodeURIComponent(branchName2);
+    }
 }
```

Comment:
Seems like the `Branch` and `Pipeline` classes should move to the new `blueocean-core-js` package?